static/app/utils/url/urlParamBatchContext.tsx

```diff
@@ -27,9 +26,9 @@ export function UrlParamBatchProvider({children}: {children: React.ReactNode}) {
 
     navigate(
       {
-        ...location,
+        pathname: window.location.pathname,
```

Comment:
This hook doesn't work when we use `location` from `useLocation` and pass it as a dependency. 

It reinstantiates the `flushUpdates` function every time location.query updates (which is exactly what we are trying to do here, we are trying to sync local state with location.query). Since we call the memoized `flushUpdates` method below using a debounce, re-instantiating prevents the debounce from firing. 

Using window.location.search here instead. 

I fixed tests accordingly.